usaspending_api/disaster/management/sql/spark/disaster_covid19_file_b.py

```diff
@@ -0,0 +1,145 @@
+INNER JOIN global_temp.ref_program_activity ON (financial_accounts_by_program_activity_object_class.program_activity_id = ref_program_activity.id)
+INNER JOIN global_temp.object_class ON (financial_accounts_by_program_activity_object_class.object_class_id = object_class.id)
+LEFT OUTER JOIN global_temp.toptier_agency ON (treasury_appropriation_account.funding_toptier_agency_id = toptier_agency.toptier_agency_id)
+LEFT OUTER JOIN global_temp.cgac AS cgac_aid ON (treasury_appropriation_account.agency_id = cgac_aid.cgac_code)
+LEFT OUTER JOIN global_temp.cgac AS cgac_ata ON (treasury_appropriation_account.allocation_transfer_agency_id = cgac_ata.cgac_code)
+WHERE (
+    disaster_emergency_fund_code.group_name = 'covid_19'
+    AND submission_attributes.is_final_balances_for_fy = TRUE
+)
+GROUP BY
+    financial_accounts_by_program_activity_object_class.financial_accounts_by_program_activity_object_class_id,
+    CASE
+        WHEN submission_attributes.quarter_format_flag = True THEN
+            CONCAT('FY', EXTRACT(YEAR FROM (financial_accounts_by_program_activity_object_class.reporting_period_end) + INTERVAL 3 months), 'Q', EXTRACT(QUARTER FROM (financial_accounts_by_program_activity_object_class.reporting_period_end) + INTERVAL 3 months))
+        ELSE
+            CONCAT('FY', EXTRACT(YEAR FROM (financial_accounts_by_program_activity_object_class.reporting_period_end) + INTERVAL 3 months), 'P', LPAD(STRING(EXTRACT(MONTH FROM (financial_accounts_by_program_activity_object_class.reporting_period_end) + INTERVAL 3 months)), 2, '0'))
+    END,
+    cgac_aid.agency_name,
+    cgac_ata.agency_name,
+    toptier_agency.name,
+    submission_attributes.reporting_agency_name,
+    treasury_appropriation_account.allocation_transfer_agency_id,
+    treasury_appropriation_account.agency_id,
+    treasury_appropriation_account.beginning_period_of_availability,
+    treasury_appropriation_account.ending_period_of_availability,
+    treasury_appropriation_account.availability_type_code,
+    treasury_appropriation_account.main_account_code,
+    treasury_appropriation_account.sub_account_code,
+    treasury_appropriation_account.tas_rendering_label,
+    treasury_appropriation_account.account_title,
+    treasury_appropriation_account.budget_function_title,
+    treasury_appropriation_account.budget_subfunction_title,
+    federal_account.federal_account_code,
+    federal_account.account_title,
+    disaster_emergency_fund_code.code,
+    disaster_emergency_fund_code.public_law,
+    disaster_emergency_fund_code.title,
+    ref_program_activity.program_activity_code,
+    ref_program_activity.program_activity_name,
+    object_class.object_class,
+    object_class.object_class_name,
+    object_class.direct_reimbursable,
+    financial_accounts_by_program_activity_object_class.obligations_incurred_by_program_object_class_cpe,
```

Comment:
I see that columns starting from this line onward were not included in the original group by. What's the reason for adding them here?

Reply:
Anytime a column is functionally dependent on the PK in PSQL, PSQL automatically adds it to the group (behind the scenes) and allows the column to be in the SELECT statement event though it isn't present in the GROUP BY. Spark does not do that. So the problem was that these columns couldn't be selected in spark.

Reply:
Interesting. I did not know that